src/shims/posix/fs.rs

```diff
@@ -383,7 +410,10 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
             let fh = &mut this.machine.file_handler;
             let (file_result, writable) = match fh.handles.get(&fd) {
-                Some(FileHandle { file, writable }) => (file.try_clone(), *writable),
+                Some(file_descriptor) => match file_descriptor.as_file_handle() {
+                    Ok(FileHandle { file, writable }) => (file.try_clone(), *writable),
+                    Err(_) => return this.handle_not_found(),
```

Comment:
This is fishy -- there are very few situations where it is a good idea to inspect an `InterpResult` with `match`.

But also, the error is wrong. The handle *was* found, it just did not support this operation. So IMO it would be reasonable to just do `file_descriptor.as_file_handle()?` here.

Reply:
Oh and please add a FIXME saying that in the future, we'd of course like to support "dup" for *all* FDs, not just the ones with an underlying `FileHandle`.